src/UI/Utils/useIsDirty.tsx

```diff
@@ -0,0 +1,99 @@
+import { useEffect, useState } from "react";
+import { InstanceAttributeModel } from "@/Core";
+
+/**
+ * Continously check if any properties in given object are modified comparing to baseToCompare param or populated with any non-default data(empty string or null)
+ * return boolean value depending on the outcome and function to override Hooks State manually
+ *
+ * @param  object
+ * @param baseToCompare
+ *
+ */
+export default function useIsDirty(
+  object,
+  baseToCompare: InstanceAttributeModel | undefined = undefined
+) {
+  const [isDirty, setIsDirty] = useState(false);
+  const overrideState = (value: boolean) => {
+    setIsDirty(value);
+  };
+  useEffect(() => {
```

Comment:
I'm gonna ask more explanation on the reasoning of this complicated sequence, please